sdks/java/io/hbase/src/main/java/org/apache/beam/sdk/io/hbase/HBaseIO.java

```diff
@@ -198,6 +201,13 @@ public Read withKeyRange(byte[] startRow, byte[] stopRow) {
       return withKeyRange(keyRange);
     }
 
+    @VisibleForTesting
+    Read useSdf() {
+      Read read = new Read(serializableConfiguration, tableId, serializableScan);
+      read.useSDF = true;
```

Comment:
Would prefer to stick to completely immutable classes. Maybe convert this one to AutoValue while we're at it?

Reply:
For the moment I think we are ok as it is, the arguments of HBaseIO are pretty small to add all the overhead of AutoValue.